app/admin/quests/create/page.tsx

```diff
@@ -0,0 +1,1049 @@
+"use client";
+
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import styles from "@styles/admin.module.css";
+import { useRouter } from "next/navigation";
+import { AdminService } from "@services/authService";
+import ProgressBar from "@components/quiz/progressBar";
+import InputCard from "@components/admin/InputCard";
+import { MenuItem, Select, SelectChangeEvent, Switch } from "@mui/material";
+import Button from "@components/UI/button";
+import {
+  QuestDefault,
+  TOKEN_ADDRESS_MAP,
+  TOKEN_DECIMAL_MAP,
+} from "@constants/common";
+import { getCurrentNetwork } from "@utils/network";
+import Textinput from "@components/admin/Textinput";
+import Dateinput from "@components/admin/DateInput";
+import QuizControls from "@components/quiz/quizControls";
+import {
+  CATEGORY_OPTIONS,
+  TASK_OPTIONS,
+  TWITTER_OPTIONS,
+  boostDefaultInput,
+  nft_uri,
+  questDefaultInput,
+  QuizQuestionDefaultInput,
+  getDefaultValues,
+} from "@constants/admin";
+import { CreateQuest, NFTUri } from "../../../../types/backTypes";
+import AdminQuestDetails from "@components/admin/QuestDetails";
+import { useNotification } from "@context/NotificationProvider";
+import Dropdown from "@components/UI/dropdown";
+
+// Define discriminated union types
+type StepMap =
+  | { type: "Quiz"; data: QuizInputType }
+  | { type: "TwitterFw"; data: TwitterFwInputType }
+  | { type: "TwitterRw"; data: TwitterRwInputType }
+  | { type: "Discord"; data: DiscordInputType }
+  | { type: "Custom"; data: CustomInputType }
+  | { type: "None"; data: object }
+  | { type: "Domain"; data: DomainInputType };
+
+export default function Page() {
+  const network = getCurrentNetwork();
+  const router = useRouter();
+  const [currentPage, setCurrentPage] = useState(1);
+  const [questId, setQuestId] = useState<number>(0);
+  const [questInput, setQuestInput] = useState<CreateQuest>(questDefaultInput);
+  const [nfturi, setNftUri] = useState<NFTUri>(nft_uri);
+  const [headingText, setHeadingText] = useState("Set up");
+  const [showBoost, setShowBoost] = useState(false);
+  const [boostInput, setBoostInput] = useState(boostDefaultInput);
+  const [startTime, setStartTime] = useState("");
+  const [endTime, setEndTime] = useState("");
+  const [showTwitterOption, setShowTwitterOption] = useState(-1);
+  const [steps, setSteps] = useState<StepMap[]>([
+    {
+      type: "None",
+      data: {},
+    },
+  ]);
+  const { showNotification } = useNotification();
+  const [finalQuestData, setFinalQuestData] =
+    useState<typeof QuestDefault>(QuestDefault);
+  const [buttonLoading, setButtonLoading] = useState(false);
+
+  const isButtonDisabled = useMemo(() => {
+    if (currentPage === 1) {
+      return (
+        !questInput.name ||
+        !questInput.desc ||
+        !questInput.start_time ||
+        !questInput.expiry ||
+        !questInput.category
```

Comment:
use a const with a naming that shows what is this standing for (what case is this giant ternary representing).

Easier to read :)

Reply:
done